examples/internal/integration/integration_test.go

```diff
@@ -1528,7 +1528,7 @@ func TestNotImplemented(t *testing.T) {
 		return
 	}
 
-	if got, want := resp.StatusCode, http.StatusNotImplemented; got != want {
+	if got, want := resp.StatusCode, http.StatusMethodNotAllowed; got != want {
```

Comment:
The fact that you needed to change this tells us that this is a potentially breaking change in behaviour. I think we need to make this behaviour configurable, maybe we only expose the error and let the user create a custom routing error handler?